src/util/Injected.js

```diff
@@ -106,6 +351,71 @@ exports.LoadUtils = () => {
 
     };
 
+    window.WWebJS.prepareMessageButtons = (buttonsOptions) => {
+        const returnObject = {};
+        if (!buttonsOptions.buttons) {
+            return returnObject;
+        }
+        
+        returnObject.title = buttonsOptions.title;
+        returnObject.footer = buttonsOptions.footer;
+    
+        if (buttonsOptions.useTemplateButtons) {
+            returnObject.isFromTemplate = true;
+            returnObject.hydratedButtons = buttonsOptions.buttons;
+            returnObject.buttons = new window.Store.TemplateButtonCollection();
+
+            returnObject.buttons.add(
+                returnObject.hydratedButtons.map((button, index) => {
+                    const i = `${null != button.index ? button.index : index}`;
+                      
+                    if (button.urlButton) {
+                        return new window.Store.TemplateButtonModel({
+                            id: i,
+                            displayText: button.urlButton?.displayText,
+                            url: button.urlButton?.url,
+                            subtype: 'url',
+                        });
+                    }
+            
+                    if (button.callButton) {
+                        return new window.Store.TemplateButtonModel({
+                            id: i,
+                            displayText: button.callButton.displayText,
+                            phoneNumber: button.callButton.phoneNumber,
+                            subtype: 'call',
+                        });
+                    }
+            
+                    return new window.Store.TemplateButtonModel({
+                        id: i,
+                        displayText: button.quickReplyButton?.displayText,
+                        selectionId: button.quickReplyButton?.id,
+                        subtype: 'quick_reply',
+                    });
+                })
+            );
+        }
+        else {
+            returnObject.isDynamicReplyButtonsMsg = true;
+
+            returnObject.dynamicReplyButtons = buttonsOptions.buttons.map((button, index) => ({
+                buttonId: button.quickReplyButton.id.toString() || `${index}`,
```

Comment:
the new error is here id undefind
buttonId: button.quickReplyButton?.id?.toString() || `${index}`,

Reply:
any update here ?

Reply:
Any update about List, and button ?